src/common/components/SelectField.jsx

```diff
@@ -40,20 +41,58 @@ const SelectField = ({
     return (
       <FormControl fullWidth={fullWidth}>
         {multiple ? (
-          <>
-            <InputLabel>{label}</InputLabel>
-            <Select
-              label={label}
+          searchable ? (
+            // Use Autocomplete for multiple with search
+            <Autocomplete
               multiple
-              value={value}
-              onChange={onChange}
-            >
-              {items.map((item) => (
-                <MenuItem key={keyGetter(item)} value={keyGetter(item)}>{titleGetter(item)}</MenuItem>
-              ))}
-            </Select>
-          </>
+              size="small"
+              options={items}
+              getOptionLabel={(option) => titleGetter(option)}
+              renderOption={(props, option) => (
+                <MenuItem {...props} key={keyGetter(option)} value={keyGetter(option)}>
+                  {titleGetter(option)}
+                </MenuItem>
+              )}
+              isOptionEqualToValue={(option, value) => keyGetter(option) === keyGetter(value)}
+              value={items.filter(item => value.includes(keyGetter(item)))}
+              onChange={(_, newValue) => {
+                const selectedIds = newValue.map(item => keyGetter(item));
+                onChange({ target: { value: selectedIds } });
+              }}
+              renderInput={(params) => (
+                <TextField {...params} label={label} placeholder={`Search... (${value.length} selected)`} />
```

Comment:
It's a bit odd to have both "search..." and "x selected". Maybe we show search when empty only. And selected if something selected. Also don't forget about localization. For selected maybe we can just show the comma-separated list.